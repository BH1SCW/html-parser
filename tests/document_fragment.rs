use html_parser::{HtmlParser, Result};
use insta::assert_debug_snapshot;

#[test]
fn it_can_parse_single_div_as_fragment() -> Result<()> {
    let markup = "<div/>";
    let ast = HtmlParser::parse(markup)?;
    assert_debug_snapshot!(ast);
    Ok(())
}
#[test]
fn it_can_parse_single_text_as_fragment() -> Result<()> {
    let markup = "hello";
    let ast = HtmlParser::parse(markup)?;
    assert_debug_snapshot!(ast);
    Ok(())
}
#[test]
fn it_can_parse_text_comment_element_as_fragment() -> Result<()> {
    let markup = "hello<!--world?--><div/>";
    let ast = HtmlParser::parse(markup)?;
    assert_debug_snapshot!(ast);
    Ok(())
}
#[test]
fn it_error_when_body_is_used_in_fragment_root() {
    let markup = "<div></div><body></body>";
    assert!(HtmlParser::parse(markup).is_err());
}
#[test]
fn it_error_when_head_is_used_in_fragment_root() {
    let markup = "<div></div><head></head>";
    assert!(HtmlParser::parse(markup).is_err());
}
#[test]
fn it_error_when_html_is_used_in_fragment_root() {
    let markup = "<div></div><html></html>";
    assert!(HtmlParser::parse(markup).is_err());
}
