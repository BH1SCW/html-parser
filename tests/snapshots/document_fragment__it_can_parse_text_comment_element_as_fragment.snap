---
source: tests/document_fragment.rs
expression: ast
---
{
  "treeType": "documentFragment",
  "nodes": [
    {
      "text": "hello"
    },
    {
      "element": {
        "name": "div",
        "variant": "void",
        "attributes": {},
        "nodes": []
      }
    }
  ]
}
